src/Bicep.Core/Semantics/JsonObjectParser.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.TypeSystem;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Bicep.Core.Semantics
+{
+    public class JsonObjectParser : ObjectParser
+    {
+        override public JToken ExtractTokenFromObject(string fileContent)
+            => fileContent.TryFromJson<JToken>();
```

Comment:
What happens here if the file content isn't valid JSON?

Reply:
it returns null, and we added a test to verify the ErrorDiagnostic is properly triggered.

The yaml case behaved a little differently and we had to add a try/catch.